db-cleanup/airflow-db-cleanup.py

```diff
@@ -91,23 +95,35 @@ def cleanup_function(**context):
 
     airflow_db_model = context["params"].get("airflow_db_model")
     age_check_column = context["params"].get("age_check_column")
+    keep_last_run = context["params"].get("keep_last_run")
+    dag_id = context["params"].get( "dag_id" )
 
     logging.info("Configurations:")
     logging.info("max_date:                 " + str(max_date))
     logging.info("enable_delete:            " + str(ENABLE_DELETE))
     logging.info("session:                  " + str(session))
     logging.info("airflow_db_model:         " + str(airflow_db_model))
     logging.info("age_check_column:         " + str(age_check_column))
+    logging.info("keep_last_run:            " + str(keep_last_run))
+    logging.info("dag_id:                   " + str(dag_id))
     logging.info("")
 
     logging.info("Running Cleanup Process...")
+    if keep_last_run:
+        query = session.query(airflow_db_model).options(load_only(age_check_column)).filter(
+            age_check_column.notin_(session.query( func.max( age_check_column)).group_by(
+                dag_id)), and_(age_check_column <= max_date))
+    else:
+        query=session.query(airflow_db_model)
+        query=query.options(load_only(age_check_column))
+        query=query.filter(age_check_column <= max_date,)
 
-    entries_to_delete = session.query(airflow_db_model).filter(
-        age_check_column <= max_date,
-        ).all()
+    entries_to_delete = query.all()
+    logging.info( query )
```

Comment:
Provide some context in the logging statement. Example:
`logging.info("query: " + str(query))`
provide better label